packages/app/src/components/CompleteUserRegistrationForm.tsx

```diff
@@ -1,14 +1,21 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import { withTranslation } from 'react-i18next';
-import loggerFactory from '~/utils/logger';
+import React, { useState, useEffect } from 'react';
+import { WithTranslation, withTranslation } from 'react-i18next';
+import axios from '~/utils/axios';
 import { withUnstatedContainers } from './UnstatedUtils';
 import AppContainer from '~/client/services/AppContainer';
 
-const logger = loggerFactory('growi:completeUserRegistration');
+interface Props {
+  t: any, //  i18next
+  appContainer: AppContainer,
```

Comment:
You can use `withUnstatedContainers` only against to class components.

`withUnstatedContainers` はクラスコンポーネントに対してのみ使うことができます。


So at first, you have to create wrapper class component and inject AppContainer into it,, then render `CompleteUserRegistrationForm: React.FC` instance and then give appContainer instance via props.

なので、まずは新しくクラスコンポーネントを作り、それに対して AppContainer を inejct する必要があります。そしてその中で `CompleteUserRegistrationForm: React.FC` をレンダリングし、props を介して appContainer インスタンスを渡してください。

Reply:
Remove appContainer from Props. It seems to be unused.

Reply:
Oh sorry, I was wrong. I noticed appContainer is used to get "csrfToken".

So, please use `apiv3Post` instead of submitting form so that you can remove appContainer from this file.